File: src/main/java/com/gmail/borlandlp/minigamesdtools/gun/BulletCreatorHub.java
package com.gmail.borlandlp.minigamesdtools.gun;

import com.gmail.borlandlp.minigamesdtools.creator.AbstractDataProvider;
import com.gmail.borlandlp.minigamesdtools.creator.CreatorHub;
import com.gmail.borlandlp.minigamesdtools.gun.bullet.GhostBullet;

public class BulletCreatorHub extends CreatorHub {
    public GhostBullet createBullet(String ID, AbstractDataProvider dataProvider) throws Exception {
        return (GhostBullet) this.create(ID, dataProvider);
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/gun/BulletCreatorHub.kt
package com.gmail.borlandlp.minigamesdtools.gun

import com.gmail.borlandlp.minigamesdtools.creator.AbstractDataProvider
import com.gmail.borlandlp.minigamesdtools.creator.CreatorHub
import com.gmail.borlandlp.minigamesdtools.gun.bullet.GhostBullet

class BulletCreatorHub : CreatorHub() {
    @Throws(Exception::class)
    override fun create(itemID: String, dataProvider: AbstractDataProvider): GhostBullet {
        return super.create(itemID, dataProvider) as GhostBullet
    }
}